src/Generating/FinishGen.cpp

```diff
@@ -665,51 +665,106 @@ void cFinishGenVines::GenFinish(cChunkDesc & a_ChunkDesc)
 
 
 
-
-
 ////////////////////////////////////////////////////////////////////////////////
 // cFinishGenSprinkleFoliage:
 
-bool cFinishGenSprinkleFoliage::TryAddSugarcane(cChunkDesc & a_ChunkDesc, int a_RelX, int a_RelY, int a_RelZ)
+bool cFinishGenSprinkleFoliage::TryAddCactus(cChunkDesc &a_ChunkDesc, int a_RelX, HEIGHTTYPE & a_RelY, int a_RelZ)
 {
-	// We'll be doing comparison to neighbors, so require the coords to be 1 block away from the chunk edges:
-	if (
-		(a_RelX < 1) || (a_RelX >= cChunkDef::Width  - 1) ||
-		(a_RelY < 1) || (a_RelY >= cChunkDef::Height - 2) ||
-		(a_RelZ < 1) || (a_RelZ >= cChunkDef::Width  - 1)
-	)
+	if (!IsDesertVariant(a_ChunkDesc.GetBiome(a_RelX, a_RelZ)))
 	{
 		return false;
 	}
 
-	// Only allow dirt, grass or sand below sugarcane:
-	switch (a_ChunkDesc.GetBlockType(a_RelX, a_RelY, a_RelZ))
+	for (int i = 0; i < 1 + (m_Noise.IntNoise2DInt(a_RelX, a_RelZ) % m_MaxCactusHeight); i++)
 	{
-		case E_BLOCK_DIRT:
-		case E_BLOCK_GRASS:
-		case E_BLOCK_SAND:
+		if (
+			(a_RelX < 1) || (a_RelX >= cChunkDef::Width - 1) ||
+			(a_RelY < 1) || (a_RelY >= cChunkDef::Height - 2) ||
+			(a_RelZ < 1) || (a_RelZ >= cChunkDef::Width - 1)
+		)
 		{
-			break;
+			return (i == 0) ? false : true;
```

Comment:
Well, I didn't remember having been drunk the evening I coded this... will remove ! Although I think putting a comment would be more explicit and avoid creating a variable for readability ?

Reply:
I prefer code, because comments get more easily out of sync.
In your case, the variable even reduces the repetition of `i != 0` (and maybe the repetition of the comment explaining this).

But this is just a personal preference. You can also use a comment.